packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/SubjectGroupViewerContainer.js

```diff
@@ -39,15 +45,41 @@ function storeMapper (stores) {
   const cellStyle = viewerConfig.cell_style || DEFAULT_CELL_STYLE
   const gridColumns = viewerConfig.grid_columns || DEFAULT_GRID_COLUMNS
   const gridRows = viewerConfig.grid_rows || DEFAULT_GRID_ROWS
+  
+  const {
+    activeStepTasks
+  } = stores.classifierStore.workflowSteps
+  const [currentTask] = activeStepTasks.filter(task => task.type === 'subjectGroupComparison')
+  
+  const {
+    addAnnotation,
+    active: classification,
+  } = stores.classifierStore.classifications
+  
+  const isCurrentTaskValidForAnnotation = !!currentTask
+
+  // Note: the Task's Annotations are initialised by the SubjectGroupComparisonTask
+  // component. However, do note that it's possible to have a
+  // SubjectGroupViewer without a SubjectGroupComparisonTask.
+  
+  // Currently only supports 1 Subject Group Task in the workflow.
+  const allAnnotations = classification ? Array.from(classification.annotations.values()) : []
+  const annotation = allAnnotations.find(item => (getType(item).name === 'SubjectGroupComparisonAnnotation'))
```

Comment:
If you're just starting the task, then `classification.annotations` won't contain an annotation for this task yet. So `annotation` might be undefined.

Reply:
Comparing this with the drawing task code, maybe use `.filter` instead of `.find` here in case there's more than one subject group task in the workflow?

Reply:
re: use of `.find` -

This line originally used a variant of the `.filter` code from InteractionLayer (`const [annotation] = allAnnotations.filter(item => (getType(item).name === 'SubjectGroupComparisonAnnotation'))`, to be exact) but I thought `.find` would be easier to read since I was purposely constraining the component to 1 Subject Group Comparison Task per workflow. I might need to rethink this if `.filter` makes more sense to others.

(Note: supporting multiple possible Subject Group X Tasks would require a mild rethink of the UI - e.g. colour markings - and I didn't want to jump into that before I had a proper use case.)